wrappers/services.go

```diff
@@ -59,25 +59,55 @@ func generateSnapServiceFile(app *snap.AppInfo) (string, error) {
 }
 
 // StartSnapServices starts service units for the applications from the snap which are services.
-func StartSnapServices(s *snap.Info, inter interacter) error {
+func StartSnapServices(s *snap.Info, inter interacter) (err error) {
+	hasReloaded := false
+	sysd := systemd.New(dirs.GlobalRootDir, inter)
+	var enabled []*snap.AppInfo
+	var started []*snap.AppInfo
+
+	defer func() {
```

Comment:
Yes, that's something like it. I was thinking about making enough helpers that the defer func(){...}() would just be a one-liner so the flow would be more continuous. In any case this is a non-binding suggestion so you have my +1 already :)